src/database/operations/update/updateBaseField.ts

```diff
@@ -0,0 +1,35 @@
+import { db } from '../../db';
+import { NotFoundError } from '../../../errors';
+import type {
+  BaseField,
+  BaseFieldWrite,
+} from '../../../types';
+
+export const updateBaseField = async (
+  id: number,
+  updateValues: BaseFieldWrite,
+): Promise<BaseField> => {
+  const {
+    label,
+    description,
+    shortCode,
+    dataType,
+  } = updateValues;
+  const result = await db.sql<BaseField>(
+    'baseFields.updateById',
+    {
+      id,
+      label,
+      description,
+      shortCode,
+      dataType,
+    },
```

Comment:
I wanted to point this pattern out -- I'd argue we're making a mistake in quite a few places by passing the `req.body` directly to the deb.sql call, unfiltered.  This is less explicit, but also gives end users the ability to inject more than what the query actually needs.  I am pretty sure those values get filtered by our tools, but it's safer to pull and pass exactly what we need.

Assuming this pattern makes it through review I'd plan to start moving our other endpoints to this new `operations` model, and when I do that I'll be correcting the mistake at the same time.

Reply:
I'm happy to implement this -- though I admit I think there is cost / increased chance of another kind of bug by taking ordered parameters that could easily be passed out of order.

For instance, there is no type checking that will catch if I accidentally pass:

```
await updateBaseField(
      id,
      body.label,
      body.description,
      body.shortCode,
    );
```
vs
```
await updateBaseField(
      id,
      body.description,
      body.label,
      body.shortCode,
    );
```

Reply:
Yeah, there's trade-offs for sure, and you're quite right that long lists of variables are hard to work with.

> Would it feel less smelly if instead of passing body the variable was named `baseFieldValues`?

Perhaps! Passing around a `req.body` feels dangerous, even though validating its contents makes it not - maybe that's just a heuristic I need to reconsider.

As I said, I don't think this is a blocker and I trust your judgement on this, and I'm fine with leaving this as-is! Thanks for talking through it with me.

Reply:
(I thought I wrote this but it isn't showing up so writing it again...)

I went with what might be the worst of all worlds, but at least addresses the two bug related concerns: the parameter to the update{entity} operation still takes the updated entity values object.  However, that object is now being explicitly constructed from `body` in the update handler, rather than the body itself being passed.